core/jwt_authentication.py

```diff
@@ -29,6 +31,11 @@ def authenticate(self, request):
         except (jwt.PyJWTError, JSONWebTokenError) as exc:
             raise exceptions.AuthenticationFailed(str(exc)) from exc
 
+        if CoreConfig.is_valid_health_facility_contract_required:
+            if (hasattr(user, 'health_facility') and hasattr(user.health_facility, 'contract_end_date') and
```

Comment:
I don't like this check as check is depending on user properties and not type of the user. For instance: 
lets say we have user
- Contractor
- Admin  

Contractor's login possibility should be determined by contract date (and it should be mandatory input). For Admin contract shouldn't matter. If contractor don't have hf assigned ti them the contract will never expire. And if system Admin for any reason is assigned to HF but has control on more HFs in system then they will not be able to log in. Not sure if that is actual use case but it seems possible. 

I think it should be more like: 
`if user.is_contractor() and user.hf.valid_contract()` would be better check. Or even better something like `user.can_login()` which could be more general approach for this. Maybe I'm overthinking this, but let me know your thoughts.

Reply:
The problem is that they want to block it for all users assigned to HF. As I understand this, some general country-wide openIMIS admin should not be assigned to any HF at all. And if you are employee of any kind in the given HF you should not have access to the system after you contract expires.

Reply:
Make sense, in that case we can keep it